### ISSUE TITLE
New IPsec Connections GUI: RADIUS servers not added to the strongSwan configuration

### ISSUE THREAD

**Issue by hostalp** [opnsense/core#6602]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [X] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [X] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

I'm unable to get the EAP-RADIUS IPsec VPN authentication to work via the new Connections configuration pages.

- The old IPsec configuration has the following options:
**VPN: IPsec: Tunnel Settings: Phase 1 / Phase 1 proposal (Authentication)**
**Authentication method**: EAP-RADIUS
**Radius servers**: _(select servers defined in System: Access: Servers)_
\
Setting these resulted in the selected RADIUS servers being added to `/usr/local/etc/strongswan.conf`

- The new Connections configuration has the following option:
**VPN: IPsec: Connections [new]: Connections: Remote**
**Authentication**: EAP RADIUS
\
And setting this sets only the authentication method in the connection configuration, but no RADIUS server is added anywhere to the strongSwan configuration which results in non-functional configuration.

Am I missing something or is there currently no way to add RADIUS servers to the configuration via the GUI?

**Expected behavior**

If EAP RADIUS authentication method is selected, RADIUS servers should be added to the strongSwan configuration.

**Describe alternatives you considered**

Configuration snippets (/usr/local/etc/strongswan.opnsense.d/*.conf) may be used as a workaround.

**Relevant log files**
```
00[CFG] loaded 0 RADIUS server configurations
14[IKE] <conn-id|1> loading EAP_RADIUS method failed
```

**Environment**

Software version used and hardware type if relevant, e.g.:

OPNsense 23.1.9 (amd64, OpenSSL).
Intel(R) Xeon(R) Gold 6346 CPU @ 3.10GHz (2 cores, 2 threads)
Network VirtIO

**Comment by AdSchellevis** [comment]
If I’m not mistaken RADIUS currently only functions in legacy mode due to how 
`ipsec_write_strongswan_conf()` collects the servers. I do need to look how this should work for both options (combined).

**Comment by AdSchellevis** [comment]
@hostalp if you are still interested, I've added a eap-radius server selection in the mobile clients screen (https://github.com/opnsense/core/commit/2c7fc072b1710addc3ac8b99a47ba407aed3a13f). Since radius is not bound to a specific connection, it's only visible when the legacy mode doesn't use radius.

**Comment by miken32** [comment]
> @hostalp if you are still interested, I've added a eap-radius server selection in the mobile clients screen ([2c7fc07](https://github.com/opnsense/core/commit/2c7fc072b1710addc3ac8b99a47ba407aed3a13f)). Since radius is not bound to a specific connection, it's only visible when the legacy mode doesn't use radius.

I just came across this problem; it's not at all intuitive for users migrating from legacy connections to new ones. I had disabled my old connection while trying to set up the new one; turns out I had to completely delete it before I could get the new one working at all! If it's not technically possible to support both, a warning should be given when you try to select EAP RADIUS authentication on a new connection with a legacy connection still active.

**Comment by AdSchellevis** [comment]
@miken32 I agree the current construction has downsides when migrating from legacy. There are some options on the "VPN: IPsec: Mobile Clients" page we could consider, if you open a ticket, I'll try to take a look when I can.

**Comment by malhal** [comment]
Important: any changes made to the eap-radius server configuration in the mobile clients screen require clicking the restart button at the top right to take effect for the next VPN connection attempt.
